### DISCUSSION TITLE
Non-virtual Adapter VDA5050 - how to use it?

### DISCUSSION THREAD

**Discussion by mclSaxony** [openTCS/opentcs#165] [Q&A]
After some struggeling with the openTCS Integration Project in regards of gradlew (_jdk v22 issue with gradle 8.7 +  proxy-trouble_) (  gradle was able to compile openTCS. 
The produced binaries are able to run. When I look inside the output-folder I see a lot of jars. Also I see opentcs-commadapter-vda5050-0.21.0.jar in kernel and kernel-control-center's folder which I try to use for an example communication with some real hardware **but I have no glue how to use/configure it**.
In the kernel-control center I see with "Adapter" only the virtual adapter. As there are a dropDown menu inside "Adapter" I expect something to be choosable, but there is no. 
Do I have to enable the real comadapter also anywhere else other then in "opentcs-kernel-defaults-custom.properties"  where I have filled in settings like

> commadapter.vehicle.vda5050.enabledVersions=1.1, 2.0
> commadapter.vehicle.vda5050.onOpModeChangeDoUpdateIntegrationLevel = AUTOMATIC=LEAVE_UNCHANGED, SEMIAUTOMATIC=LEAVE_UNCHANGED, SERVICE=LEAVE_UNCHANGED, MANUAL=LEAVE_UNCHANGED, TEACHIN=LEAVE_UNCHANGED

as I have read (somewhere) in the documentation?

My plan is to use MQTT to send some prepared topics towards openTCS al least to see some "ghost-vehicle" in the OperationDesk. Also I would expect that openTCS sends some stuff towards the MQTT broker which I can observe....

How to achieve this? Is there anything prepared for this USeCase?

Thanks a lot in advance

--jens

**Comment by swltr** [comment]
@mclSaxony How the VDA5050 driver is configured / intended to be used is documented in the driver project at https://github.com/openTCS/opentcs-commadapter-vda5050. Please check that documentation first, and if anything doesn't work as expected based on the documentation, please let us know!
